code/controllers/subsystem/job.dm

```diff
@@ -106,7 +106,7 @@ SUBSYSTEM_DEF(job)
 		var/datum/job/job = GetJob(rank)
 		if(!job)
 			return FALSE
-		if(is_banned_from(player.ckey, rank) || QDELETED(player))
+		if(QDELETED(player) || is_banned_from(player.ckey, rank))
```

Comment:
Potentially short circuit on a fairly quick check instead of running an expensive proc first.